src/shared/sleep-config.c

```diff
@@ -141,6 +142,147 @@ int read_battery_capacity_percentage(void) {
 }
 
+static int get_battery_identifiers(char **ret_identifiers) {
+        _cleanup_free_ char *bat_manufacturer = NULL, *bat_model_name = NULL, *bat_serial_number = NULL;
+        _cleanup_free_ char *p = NULL;
+        int r;
+
+        assert(ret_identifiers);
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/manufacturer", &bat_manufacturer);
+        if (r == -ENOENT)
+               log_debug_errno(r, "/sys/class/power_supply/BAT0/manufacturer is unavailable: %m");
+        else if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/manufacturer: %m");
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/model_name", &bat_model_name);
+        if (r == -ENOENT)
+               log_debug_errno(r, "/sys/class/power_supply/BAT0/model_name is unavailable: %m");
+        else if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/model_name: %m");
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/serial_number", &bat_serial_number);
+        if (r == -ENOENT)
+               log_debug_errno(r, "/sys/class/power_supply/BAT0/serial_number is unavailable: %m");
+        else if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/serial_number: %m");
+
+        p = strjoina(bat_manufacturer, ":", bat_model_name, ":", bat_serial_number);
+        if (!p)
+                return log_oom_debug();
+
+        *ret_identifiers = TAKE_PTR(p);
+
+        return 0;
+}
+
+static int get_system_battery_identifier_hash(char **ret_hash) {
+        _cleanup_free_ char *p = NULL, *battery_id = NULL;
+        sd_id128_t machine_id, product_id;
+        int r;
+
+        assert(ret_hash);
+
+        r = get_battery_identifiers(&battery_id);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get battery identifier: %m");
+
+        r = sd_id128_get_machine(&machine_id);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get machine ID: %m");
+
+        r = id128_get_product(&product_id);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to get product ID: %m");
+
+        p = strjoina(battery_id, ":", SD_ID128_TO_STRING(machine_id), ":", SD_ID128_TO_STRING(product_id));
+        if (!p)
+                return log_oom_debug();
+        /* add hash function when everything is working */
+        *ret_hash = TAKE_PTR(p);
+
+        return 0;
+}
+
+static int battery_discharge_rate_in_range(int battery_discharge_rate) {
+        if (battery_discharge_rate <= 0 || battery_discharge_rate >= 200)
+               return 0;
+               /* battery discharge rate is not in a valid range */
+        log_debug("Battery discharge rate is in valid range");
+        return 1;
+}
+
+int get_battery_discharge_rate(void) {
+        _cleanup_free_ char *hash_id_discharge_rate = NULL;
+        const char *p = NULL, *filepath = NULL;
```

Comment:
no need to initialize these two here, they just get overwritten

Reply:
Yeah. Okay. Removed. Thanks!